packages/react/src/useBatchPurchase.ts

```diff
@@ -0,0 +1,78 @@
+import { Signer } from '@ethersproject/abstract-signer'
+import { BigNumberish } from '@ethersproject/bignumber'
+import { TransactionHash, UUID } from '@liteflow/core'
+import { useCallback, useContext, useState } from 'react'
+import invariant from 'ts-invariant'
+import { LiteflowContext } from './context'
+import { ErrorMessages } from './errorMessages'
+
+export enum BatchPurchaseStep {
+  INITIAL,
+  TRANSACTION_SIGNATURE,
+  TRANSACTION_PENDING,
+  OWNERSHIP,
+}
+
+type batchPurchaseFn = (
+  purchases: { offerId: UUID; quantity: BigNumberish }[],
+) => Promise<void>
+
+export default function useBatchPurchase(signer: Signer | undefined): [
+  batchPurchaseFn,
+  {
+    activeStep: BatchPurchaseStep
+    transactionHash: TransactionHash | undefined
+  },
+] {
+  const { client } = useContext(LiteflowContext)
+  const [activeStep, setActiveProcess] = useState<BatchPurchaseStep>(
+    BatchPurchaseStep.INITIAL,
+  )
+  const [transactionHash, setTransactionHash] = useState<TransactionHash>()
+
+  const onProgress = useCallback(
+    (state) => {
+      switch (state.type) {
+        case 'TRANSACTION_SIGNATURE': {
+          setActiveProcess(BatchPurchaseStep.TRANSACTION_PENDING)
+          break
+        }
+        case 'TRANSACTION_PENDING': {
+          setActiveProcess(BatchPurchaseStep.TRANSACTION_PENDING)
+          setTransactionHash(state.payload.txHash)
```

Comment:
Shouldn't the transaction hash be reset when changing state?

Reply:
I see that's the same logic as https://github.com/liteflow-labs/liteflow-js/blob/5ca797516fb96ad626d5461ee2869c977653756b/packages/react/src/useAcceptOffer.ts
Let's keep the same logic then 👍

Reply:
This is following the same flow we have on other functions. It is definitely something that could be done, but it should be done across the whole library for consistency. I don't want to introduce inconsistency here.

Reply:
Actually maybe the reset should be done in the beginning of the `batchPurchase` function. It reset in the finally, but not on start.

Reply:
moved to a dedicated issue https://github.com/liteflow-labs/liteflow-js/issues/203